test/functional/test-experiments.js

```diff
@@ -577,6 +578,22 @@ describe('isCanary', () => {
   });
 });
 
+describe('getBinaryType', () => {
+  it('should return correct type', () => {
+    const win = {
+      AMP_CONFIG: {
+        type: 'production',
+      },
+    };
+    expect(getBinaryType(win)).to.equal('production');
+    win.AMP_CONFIG.type = 'canary';
+    expect(getBinaryType(win)).to.equal('canary');
+  });
+  it('should return "unknown"', () => {
+    expect(getBinaryType({})).to.equal('unknown');
```

Comment:
Maybe add a test for when `win.AMP_CONFIG` exists, but doesn't contain a `type` field? (My original suggestion would have failed this test, but your updated implementation should pass it.)